### DISCUSSION TITLE
Supported polynomial degree range in HEXL and OpenFHE

### DISCUSSION THREAD

**Discussion by gagoritsis** [intel/hexl#142] [Q&A] [answered]
Hello all,

I am new to the subject of FHE and I a started looking into OpenFHE and HEXL/HEXL-FPGA and I have some questions. 

1. I am trying to understand if the HEXL's supported maximum polynomial degree N=2^{17} is due to the maximum polynomial degree supported by libraries using an RNS representation, such as OpenFHE, or is it because of another limitation. Could you please explain why this range was selected? 
2. In Intel HEXL-FPGA the maximum polynomial degree supported is 2^{15}. Is this because of resource constraints in the FPGAs used or is there something else I am missing?
3. Regarding q, my understanding is that q\~62bits to satisfy the Harvey's butterfly requirement q<β/4, where β=2^{64} the word size. Why is it stated as "up to q\~62 bits" and not as "up to q **=** 62 bits"?

Thank you in advance!

**Comment by joserochh** [comment, marked as answer]
Hello @gagoritsis. 

For question number one. N=2^{17} is referring  to typical use case range. I don't recall any limit on that. Did you have any problem with Ns bigger than that. 
For question number two I am not aware of that.
Regarding question number three. I see it was stated like that from the very beginning of the repo. From later experience with HEXL I can only guess it was written like that because that was not a hard limit. Probably there were many instances on which q = 63 would also work, that was a situation I saw in the past.  However, if that were the case I would prefer to state it as "up to q = 62 bits".
